packages/core/src/lib/base_protocol.ts

```diff
@@ -3,6 +3,7 @@ import type { Peer, Stream } from "@libp2p/interface";
 import type {
```

Comment:
please, let's not introduce changes into this file and instead prefer peer in `store.ts` with something like
```typescript
const preferedPeer = this.libp2p.getConnections().find(c => c.remotePeer.toString() === preferedPeerIdStr);
const peer = preferedPeer || this.getPeer();
```

Reply:
aye aye, good spot!

Reply:
35a8de306b4e812704d33794e4fb6cad0c57e6ef